WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -609,7 +609,7 @@ private static void browsePostOrPageEx(
                     shareSubject,
                     true,
                     startPreviewForResult);
-        } else if (site.isJetpackConnected()) {
+        } else if (site.isJetpackConnected() && !TextUtils.isEmpty(site.getFrameNonce())) {
```

Comment:
I believe we can achieve the same thing with this:

```suggestion
        } else if (site.isJetpackConnected() && site.isUsingWpComRestApi()) {
```

Alternatively we can just add the `site.isUsingWpComRestApi()` but also keep `!TextUtils.isEmpty(site.getFrameNonce())`.

I think all 3 of these solutions would work, I just like the explicit `isUsingWpComRestApi` check because it makes the code a bit clearer for me since we call `openJetpackBlogPostPreview` afterwards.

Reply:
It crossed my mind that `isUsingWpComRestApi` is identical to `frameNonce != empty`, but I wasn't sure. 👍 Definitely a nice improvement. Thanks